app/jobs/bag_move_job.rb

```diff
@@ -4,68 +4,63 @@
 
 class BagMoveJob < ApplicationJob
 
-  def perform(queue_item, errors: [], validator: nil)
+  def perform(queue_item)
     @queue_item = queue_item
-    @src_path = queue_item.package.src_path
-    @dest_path = queue_item.package.dest_path
-    @errors = errors
+    @package = queue_item.package
+    @errors = []
 
-    # TODO: Pull this up and out
-    # TODO: Use Services.storage.create
-    if validator.nil? && File.exist?(src_path)
-      validator = Chipmunk::Bag::Validator.new(
-        queue_item.package,
-        errors,
-        Chipmunk::Bag.new(src_path)
-      )
-    end
-
-    begin
-      # TODO
-      #  - if all validation succeeds:
-      #    - start a transaction that updates the request to complete
-      #    - move the bag into place
-      #    - success: commit the transaction
-      #    - failure (exception) - transaction automatically rolls back
+    # TODO
+    #  - if all validation succeeds:
+    #    - start a transaction that updates the request to complete
+    #    - move the bag into place
+    #    - success: commit the transaction
+    #    - failure (exception) - transaction automatically rolls back
 
-      validate_with(validator)
-    rescue StandardError => exception
-      errors << "#{exception.backtrace.first}: #{exception.message} (#{exception.class})"
-      errors << exception.backtrace.drop(1).map {|s| "\t#{s}" }
-      record_failure(errors)
-      raise exception
-    end
+    ingest! or record_failure
   end
 
   private
 
-  attr_accessor :queue_item, :src_path, :dest_path, :errors
+  def ingest!
+    validate and move_bag and record_success
+  rescue StandardError => error
+    log_exception(error)
+  end
 
-  def validate_with(validator)
-    if validator.valid?
-      FileUtils.mkdir_p(File.dirname(dest_path))
-      File.rename(src_path, dest_path)
-      record_success
-    else
-      record_failure(errors)
-    end
+  def validate
+    package.valid_for_ingest?(errors)
```

Comment:
I know `valid_for_ingest?` is in a temporary shape at the moment. I don't really like the pattern of passing an array mutating it directly - perhaps this was originally a placeholder for passing a logger? We shouldn't try to fix this right now, but I think it's worth recording as a smell to fix, especially as the mutating errors array starts being passed around more.

Reply:
Per discussion, we agree that some sort of validation or validation result object is likely to be our destination. Passing a bare array around is a bit smelly (primitive obsession).